Lib/test/datetimetester.py

```diff
@@ -390,6 +390,14 @@ def test_deepcopy(self):
         tz_copy = copy.deepcopy(tz)
         self.assertIs(tz_copy, tz)
 
+    def test_timezone_offset_close_twenty_four(self):
```

Comment:
Can you make this a more generic, parametrized test (use `self.subTest`)? I believe we should test both the negative and positive bounds (e.g. `-23:59:59.999999` and `+23:59:59.999999`).

Also, please rename the test to something like `test_offset_boundaries`.